### PR TITLE
feat: Add file check for image file when uploading models [CORE-3175]

### PR DESCRIPTION
**Pull request by JemimaBrewer**
Adds a function to check the image file type to utils.py, edits some existing tests on test_upload.py in models folder, and adds tests for the new image file function to utils.py

### CODE REVIEW

File: dafni_cli/models/upload.py [hunk 1]
```diff
@@ -34,6 +34,10 @@ def upload_model(
                                    upload a new model.
         json (bool): Whether to print the raw json returned by the DAFNI API
     """
+    if not is_valid_image_file(image_path):
+        click.echo("Invalid image file format")
```
Comment (sarahbyrnie) [inline dafni_cli/models/upload.py:38]: This message should maybe be a bit more decriptive
```suggestion
        click.echo("Your model image file type is incorrect. Please check you've entered the correct file and try again")
```

File: test/models/test_upload.py [hunk 2]
```diff
@@ -165,3 +165,30 @@ def test_model_upload_exits_for_validation_error_json(self):
         validation error with json = True"""
 
         self._test_model_upload_exits_for_validation_error(json=True)
+
+    def test_model_upload_exits_for_incorrect_image_file(self):
```
Comment (sarahbyrnie) [inline test/models/test_upload.py:169]:
```suggestion
    def test_model_upload_exits_for_incorrect_image_file_type(self):
```

File: test/models/test_upload.py [hunk 3]
```diff
@@ -165,3 +165,30 @@ def test_model_upload_exits_for_validation_error_json(self):
...
+            )
+
+        self.assertEqual(err.exception.code, 1)
+        self.mock_optional_echo.assert_not_called()
+        self.mock_validate_model_definition.assert_not_called()
+        self.mock_get_model_upload_urls.assert_not_called()
+        self.mock_upload_file_to_minio.assert_not_called()
+        self.mock_model_version_ingest.assert_not_called()
+
+        self.mock_click.echo.assert_called_once_with("Invalid image file format")
```
Comment (sarahbyrnie) [inline test/models/test_upload.py:194]: I would move this to be the first assertion you make as this is called before the other functions would be called

File: dafni_cli/models/upload.py (hunk 1 above)
Comment (joelvdavies) [inline dafni_cli/models/upload.py:38]: Could we also print out the valid file types, perhaps something like
Wrong definition file type '.json', valid file types are '.tar'

Comment (joelvdavies) [inline dafni_cli/models/upload.py:38]: Could we also print out the valid file types, perhaps something like
`Wrong definition file type '.json', valid file types are '.tar', '.tar.gz'`

File: dafni_cli/utils.py [hunk 4]
```diff
@@ -258,3 +258,21 @@ def optional_echo(string: str, should_not_print: bool):
...
+    """Returns whether a file name contains the valid file type for docker image
+
+    Args:
+        file_name (Path): name of file to be checked
+
+    Returns:
+        bool: Whether the file type is valid
+    """
+    try:
+        acceptable_file_types = ["tar"]
```
Comment (joelvdavies) [inline dafni_cli/utils.py:273]: I think `.tar.gz` should also be accepted.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Nicely done, just a few suggestions.

File: test/test_utils.py [hunk 5]
```diff
@@ -589,3 +589,40 @@ def test_does_not_output_when_passed_true(self, mock_click):
...
         mock_click.echo.assert_not_called()
+
+
+class TestIsValidImageFile(TestCase):
+    """Test class to test the is_valid_image_file function"""
+
+    def test_true_returned_if_correct_file_type_is_given(self):
+        """Tests is_valid_image_file returns true when passed a correct file type"""
+        # SETUP
+        file_name = Path("path/to/image.tar")
```
Comment (sarahbyrnie) [inline test/test_utils.py:600]: Could you add a loop in here so that this tests both valid file types

**Review by sarahbyrnie — APPROVED** [review]

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**